src/Export/ProductDebugService.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Export;
+
+use FINDOLOGIC\FinSearch\Export\Errors\ExportErrors;
+use FINDOLOGIC\FinSearch\Utils\Utils;
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+class ProductDebugService extends ProductService
+{
+    public const CONTAINER_ID = 'fin_search.product_debug_service';
+
+    /** @var string */
+    private $productId;
+
+    /** @var $shopkey */
+    private $shopkey;
+
+    /** @var ExportErrors */
+    private $exportErrors;
+
+    /** @var ProductEntity */
+    private $product;
+
+    /** @var string[] */
+    private $reasons = [];
+
+    public function getDebugInformation(string $productId, string $shopkey, ExportErrors $exportErrors): JsonResponse
```

Comment:
Maybe split this up into smaller chunks where the sub-classes have their own responsibility.

* Product fetching
* Error collecting
* Mapping data to a JsonResponse